backend/api/measure/auth.go

```diff
@@ -615,55 +628,71 @@ func RefreshToken(c *gin.Context) {
 
 	defer tx.Rollback(ctx)
 
-	userId := oldSession.UserID.String()
-
-	user := User{ID: &userId}
-	team, err := user.getOwnTeam(ctx)
+	newSession, err := authsession.NewAuthSession(oldSession.UserID, oldSession.OwnTeamId, oldSession.OAuthProvider, oldSession.UserMeta)
 	if err != nil {
-		msg := "failed to lookup user's team"
 		fmt.Println(msg, err)
-		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+		c.JSON(http.StatusInternalServerError, gin.H{
 			"error": msg,
 		})
 		return
 	}
 
-	newSession, err := authsession.NewAuthSession(oldSession.UserID, *team.ID, oldSession.OAuthProvider, oldSession.UserMeta)
-	if err != nil {
+	if err = authsession.RemoveSession(ctx, jti, &tx); err != nil {
 		fmt.Println(msg, err)
 		c.JSON(http.StatusInternalServerError, gin.H{
 			"error": msg,
 		})
 		return
 	}
 
-	if err = authsession.RemoveSession(ctx, jti, &tx); err != nil {
+	if err = newSession.Save(ctx, &tx); err != nil {
 		fmt.Println(msg, err)
 		c.JSON(http.StatusInternalServerError, gin.H{
 			"error": msg,
 		})
 		return
 	}
 
-	if err = newSession.Save(ctx, &tx); err != nil {
+	if err = tx.Commit(ctx); err != nil {
 		fmt.Println(msg, err)
 		c.JSON(http.StatusInternalServerError, gin.H{
 			"error": msg,
 		})
 		return
 	}
 
-	if err = tx.Commit(ctx); err != nil {
-		fmt.Println(msg, err)
+	authsession.SetCookies(c, newSession.AccessToken, newSession.RefreshToken,
+		newSession.AccessTokenExpiryAt, newSession.RefreshTokenExpiryAt)
+
+	c.JSON(http.StatusOK, gin.H{
+		"ok": true,
+	})
+}
+
+// GetSession returns the current session information
+func GetAuthSession(c *gin.Context) {
+	userId := c.GetString("userId")
+	ownTeamId := c.GetString("ownTeamId")
+
+	if userId == "" {
+		c.JSON(http.StatusUnauthorized, gin.H{
+			"error": "Not authenticated",
+		})
+		return
+	}
+
+	if ownTeamId == "" {
 		c.JSON(http.StatusInternalServerError, gin.H{
-			"error": msg,
+			"error": "Own team not found for userId: " + userId,
 		})
 		return
 	}
 
 	c.JSON(http.StatusOK, gin.H{
-		"access_token":  newSession.AccessToken,
-		"refresh_token": newSession.RefreshToken,
+		"user": gin.H{
+			"id":          userId,
+			"own_team_id": ownTeamId,
+		},
 	})
 }
```

Comment:
@anupcowkur What is exactly going on in this function?

Reply:
This function parses the access token from the cookie and returns a session object with user id and default team id. 
Earlier, it was being decoded client side from access token. Since client side JS does not have direct access to the access token anymore due to using http only cookies, it needs to make an api call to the server passing in the auth cookie to get access to the session object.

Reply:
It looked confusing because on the surface it seems it's just reading/validating the request and then echoing it back again. But, the client internally will call the `/auth/refresh` API when `/auth/session` is called because it will use the `fetchMeasure` proxied fetch function.